refnx/reduce/platypusnexus.py

```diff
@@ -498,6 +498,142 @@ def _chopper_values(self, h5data):
+        d["pol_flip_freq"] = h5d[
+            "entry1/instrument/polarizer_flipper/flip_frequency"
+        ][0]
+        d["pol_flip_current"] = h5d[
+            "entry1/instrument/polarizer_flipper/flip_current"
+        ][0]
+        d["pol_flip_voltage"] = h5d[
+            "entry1/instrument/polarizer_flipper/flip_voltage"
+        ][0]
+        d["pol_flip_status"] = h5d[
+            "entry1/instrument/polarizer_flipper/flip_on"
+        ][0]
+        d["pol_guide_current"] = h5d[
+            "entry1/instrument/polarizer_flipper/guide_current"
+        ][0]
+        return d
+
+    def _analyser_flippers(self, d, h5d):
+        d["anal_flip_freq"] = h5d[
+            "entry1/instrument/analyzer_flipper/flip_frequency"
+        ][0]
+        d["anal_flip_current"] = h5d[
+            "entry1/instrument/analyzer_flipper/flip_current"
+        ][0]
+        d["anal_flip_voltage"] = h5d[
+            "entry1/instrument/analyzer_flipper/flip_voltage"
+        ][0]
+        d["anal_flip_status"] = h5d[
+            "entry1/instrument/analyzer_flipper/flip_on"
+        ][0]
+        d["anal_guide_current"] = h5d[
+            "entry1/instrument/analyzer_flipper/guide_current"
+        ][0]
+        return d
+
+    def _analyser_params(self, d, h5d):
+        d["analyzer_base"] = h5d["entry1/instrument/polarizer/anal_base"][0]
+        d["analyser_dist"] = h5d["entry1/instrument/polarizer/anal_distance"][
+            0
+        ]
+        d["rotation"] = h5d["entry1/instrument/polarizer/rotation"][0]
+        d["z_trans"] = h5d["entry1/instrument/polarizer/z_translation"][0]
+        return d
+
+    def _check_sample_environments(self, d, h5d):
+        try:
+            # Try adding temperature sensor values to dict
+            d["temp_sensorA"] = h5d["entry1/sample/tc1/sensor/sensorValueA"][0]
+            d["temp_sensorB"] = h5d["entry1/sample/tc1/sensor/sensorValueB"][0]
+            d["temp_sensorC"] = h5d["entry1/sample/tc1/sensor/sensorValueC"][0]
+            d["temp_sensorD"] = h5d["entry1/sample/tc1/sensor/sensorValueD"][0]
+            d["temp_setpt1"] = h5d["entry1/sample/tc1/sensor/setpoint1"][0]
+            d["temp_setpt2"] = h5d["entry1/sample/tc1/sensor/setpoint2"][0]
+            self.iscryo = True
+        except KeyError:
+            # Temperature sensor not used in measurement - set to None
+            d["temp_sensorA"] = None
+            d["temp_sensorB"] = None
+            d["temp_sensorC"] = None
+            d["temp_sensorD"] = None
+            d["temp_setpt1"] = None
+            d["temp_setpt2"] = None
+            self.iscryo = False
+
+        try:
+            # Try adding voltage supply to dict
+            d["pow_supply_volts"] = h5d["entry1/sample/power_supply/voltage"][
+                0
+            ]
+            d["pow_supply_current"] = h5d["entry1/sample/power_supply/amps"][0]
+            d["pow_supply_relay"] = h5d["entry1/sample/power_supply/relay"][0]
+            self.isPS = True
```

Comment:
is_PS, but it's not clear what PS is. Perhaps is_power_supply?

Reply:
Yes this is correct